### DISCUSSION TITLE
Straightening image: a switch to preserve image size

### DISCUSSION THREAD

**Discussion by bulbaka** [tannerhelland/PhotoDemon#609] [Ideas]
Current behavior:
When straightening an image, either from the menu (Image - Rotate - Straighten...) or using the Measurement tool subfunction (Measurement tool > use measurement to: [straighten image] | [straighten layer]) it would always resize the image.
This is also the case for layer straightening: via the menu (Layer - Orientation - Straighten...) or the Measurement tool (same as above).

Desired behavior:
An option to preserve the original size upon image/layer straightening.

Note:
Such an option exists for both, image rotating: Image - Rotate - Rotate arbitrary... > size: [preserve] | [enlarge to fit], and layer rotating: Layer - Orientation - Rotate arbitrary... > size: [preserve] | [enlarge to fit].

It would be great to have such an option (switch) for the image/layer straightening (as done for the arbitrary rotation).

Thank you.

**Comment by tannerhelland** [comment]
Thank you for this suggestion, @bulbaka .  I want to be sure I understand correctly, so may I ask for some clarification?

Right now, PhotoDemon *always* maintains image dimensions (specifically, the image's width and height in pixels) when straightening the image.  This is what makes `Straighten` different from `Rotate`.  (I believe this is a consistent convention across most photo editors.)

Maintaining image dimensions (width and height) after straightening does require a little bit of "zooming", to erase the corners where the rotation would cause "emptiness" to appear.

It sounds like you would prefer to remove this "zoom" behavior, and instead, **reduce** the image's width and height (a "crop" action, of sorts) to solve the problem of empty corners after rotation.

Do I understand your idea correctly?
